src/main/java/org/jboss/pnc/reqour/facade/clone/GitCloneProvider.java

```diff
@@ -62,6 +62,8 @@ public void clone(RepositoryCloneRequest cloneRequest) {
         } else {
             cloneRefOnly(cloneRequest, cloneDir);
         }
+
+        return cloneRequest;
```

Comment:
because this abstraction in `TaskExecutorImpl`: `runningTasks.put(taskID, executor.supplyAsync(() -> syncExecutor.apply(request)).whenComplete(callbackSender));`
is using `supplyAsync`